### PR TITLE
Update workflows to get version from Cargo.toml

### PR DESCRIPTION
**Pull request by mikenye**
Update github workflows to utilise `get_version_method: cargo_toml_file_in_repo` from sdr-enthusiasts/common-github-workflows

### CODE REVIEW

**Comment by mikenye** [comment]
Hi Fred,

Apologies for the large number of commits - it seems neccessary when testing actions... :)

This PR will update the workflows to:

On pull to main:

- Pre-build the acars_router binaries for armv7, arm64 and amd64.
- I had to drop i386 as I couldn't get cross-build working, we can revisit in future if there's a requirement.
- I've changed the Dockerfile to pull in the pre-built binaries, rather than build them inside the image.

On push to main:

- Everything above, plus:
- Provided no issues with the build:
  - Create a tag with the version number from Cargo.toml
  - Create a release from that tag version

I've had to implement a kind of hacky workaround to determine which binary to run within the container. I'm copying all three binaries, and have modified the run script to launch whichever one works... Probably not the best method, but it works.

In future, a better method would be to determine which version works during the docker build, and only use that one.

This will likely suffice for now. :-)

File: .dockerignore [hunk 1]
```diff
@@ -4,4 +4,3 @@
 .gitattributes
 READMETEMPLATE.md
 README.md
```
Comment (fredclausen) [inline .dockerignore:6]: The target dir for the docker ignore should probably still be there for local testing of the image.

**Review by fredclausen — APPROVED** [review]
I’m being very pedantic. Looks excellent Mike.

File: .dockerignore (hunk 1 above)
Comment (mikenye) [reply .dockerignore:6]: I'm copying the binaries out of the target directory... I can probably put a step in the build process to consolidate the binaries elsewhere?

File: Dockerfile [hunk 2]
```diff
@@ -20,18 +10,28 @@ ENV AR_LISTEN_UDP_ACARS=5550 \
     AR_SERVE_ZMQ_VDLM2=45555
 
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
-COPY rootfs /
-COPY --from=builder /tmp/acars_router/target/release/acars_router /opt/acars_router
+COPY ./rootfs /
+COPY ./target/armv7-unknown-linux-gnueabihf/release/acars_router /opt/acars_router.armv7
```
Comment (mikenye) [inline Dockerfile:14]: See here

File: .dockerignore (hunk 1 above)
Comment (mikenye) [reply .dockerignore:6]: Hmm. You make a good point. This may break local testing. We may need a Dockerfile.local or another workaround.

Comment (fredclausen) [reply .dockerignore:6]: It definitely will break local testing, but your idea about a `Dockerfile.local` should fix it.

**Pull request merged**